implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -254,6 +273,22 @@ public String getRawValue(String name) {
         return configSourcesRef.get();
     }
 
+    public void unlockSecrets(Runnable runnable) {
+        SecretKeys.doUnlocked(runnable);
+    }
+
+    public <T> T unlockSecrets(Supplier<T> supplier) {
+        return SecretKeys.doUnlocked(supplier);
+    }
```

Comment:
I'm sorry I've been very unclear about all of this: it is my opinion that the doLocked/doUnlocked should only be available from one place (either `SmallRyeConfig` or `SecretKeys`) and should be `static` (because the underlying mechanism is global, so making it appear that it is bound to a configuration instance may mislead the user into thinking that only the configuration in question should be affected).

Reply:
Hum, that raises another interesting question. Since the secret keys are defined by config instance, should we consider the lock / unlock also per config instance?

Reply:
I would say "no", because from a security standpoint, it's either OK to access secrets (generally speaking) or it is not OK.  From a technical standpoint, having many thread local variables is expensive.  So I lean towards "no", it should remain global.

Reply:
Ok, let's leave that way. I'll fix the previous comment.